libs/driver-adapters/src/types.rs

```diff
@@ -27,34 +27,30 @@ pub(crate) struct JsConnectionInfo {
 
 impl JsConnectionInfo {
     pub fn into_external_connection_info(self, provider: &AdapterProvider) -> ExternalConnectionInfo {
-        let schema_name = self.get_schema_name(provider);
-        let sql_family = SqlFamily::from(provider);
-
         ExternalConnectionInfo::new(
-            sql_family,
-            schema_name.to_owned(),
+            SqlFamily::from(provider),
+            self.schema_name(provider).map(ToOwned::to_owned),
             self.max_bind_values.map(|v| v as usize),
             self.supports_relation_joins,
         )
     }
 
-    fn get_schema_name(&self, provider: &AdapterProvider) -> &str {
-        match self.schema_name.as_ref() {
-            Some(name) => name,
-            None => self.default_schema_name(provider),
-        }
+    fn schema_name(&self, provider: &AdapterProvider) -> Option<&str> {
+        self.schema_name
+            .as_deref()
+            .or_else(|| self.default_schema_name(provider))
     }
 
-    fn default_schema_name(&self, provider: &AdapterProvider) -> &str {
+    fn default_schema_name(&self, provider: &AdapterProvider) -> Option<&str> {
         match provider {
             #[cfg(feature = "mysql")]
-            AdapterProvider::Mysql => quaint::connector::DEFAULT_MYSQL_DB,
+            AdapterProvider::Mysql => None,
             #[cfg(feature = "postgresql")]
-            AdapterProvider::Postgres => quaint::connector::DEFAULT_POSTGRES_SCHEMA,
+            AdapterProvider::Postgres => Some(quaint::connector::DEFAULT_POSTGRES_SCHEMA),
             #[cfg(feature = "sqlite")]
-            AdapterProvider::Sqlite => quaint::connector::DEFAULT_SQLITE_DATABASE,
+            AdapterProvider::Sqlite => Some(quaint::connector::DEFAULT_SQLITE_DATABASE),
             #[cfg(feature = "mssql")]
-            AdapterProvider::SqlServer => quaint::connector::DEFAULT_MSSQL_SCHEMA,
+            AdapterProvider::SqlServer => Some(quaint::connector::DEFAULT_MSSQL_SCHEMA),
         }
     }
```

Comment:
I'd probably prefer to remove this logic from here completely at this point and move it to driver adapters, but I don't have time to refactor it in this sprint.